pkg/machine/e2e/stop_test.go

```diff
@@ -46,5 +48,16 @@ var _ = Describe("podman machine stop", func() {
 		Expect(err).ToNot(HaveOccurred())
 		Expect(stopAgain).To(Exit((0)))
 		Expect(stopAgain.outputToString()).To(ContainSubstring(fmt.Sprintf("Machine \"%s\" stopped successfully", name)))
+
+		// Stopping a machine should update the last up time
+		inspectJSON := new(inspectMachine)
+		inspectSession, err := mb.setName(name).setCmd(inspectJSON).run()
+		Expect(err).ToNot(HaveOccurred())
+		Expect(inspectSession).To(Exit(0))
+
+		var inspectInfo []machine.InspectInfo
+		err = jsoniter.Unmarshal(inspectSession.Bytes(), &inspectInfo)
+		Expect(err).ToNot(HaveOccurred())
+		Expect(inspectInfo[0].LastUp).To(Not(Equal("0001-01-01T00:00:00Z")))
```

Comment:
Not Equal checks are bad expect for some few exceptions were there is no better way.
The best thing would to take the time before you call stop, then parse this timestamp back into a time and then compare that the time is actually after that.

Also the whole json unmarshal thing is really not needed you can just use something like `podman machine inspect --format {{.LastUp}}`

Reply:
Updated, thanks!